docs/07-how_to_run_a_node/03-run_using_docker/04-run-validator-node.md

```diff
@@ -0,0 +1,212 @@
+```bash
+docker run --rm -ti --entrypoint nh-node horizenlabs/zkverify:latest key inspect --scheme ed25519
+```
+
+and provide same secret phrase when prompted for (`URI:`), then hit enter; you should get a response similar to:
+
+```bash
+Secret phrase:       demise trumpet minor soup worth airport minor height sauce legend flag timber
+  Network ID:        substrate
+  Secret seed:       0x9b6a3ec8063e64e9d896ed8dbcd895d7fd0d7a3a982ed9b6839e2b55c49b9e15
+  Public key (hex):  0x0dbccabf681188116e642c1dbc9332a2bbec7fbef1792196879a3cba6c52464b
+  Account ID:        0x0dbccabf681188116e642c1dbc9332a2bbec7fbef1792196879a3cba6c52464b
+  Public key (SS58): 5CNiZaphDhE8gT7cCDNZrXkd6vFfsuPjNQqdS8eEEw8mroHp
+  SS58 Address:      5CNiZaphDhE8gT7cCDNZrXkd6vFfsuPjNQqdS8eEEw8mroHp
+```
+
+the `Public key (hex)` representing the Grandpa key.
+
+Now you just need to concatenate the three keys respecting this order: Babe, Grandpa and ImOnline. In the example above this would result in: `0xc0c07abce7879c09231fcbd07165cfaabc4a634636850578a914b08b87cf99140dbccabf681188116e642c1dbc9332a2bbec7fbef1792196879a3cba6c52464bc0c07abce7879c09231fcbd07165cfaabc4a634636850578a914b08b87cf9914` (note hexadecimal prefix `0x` is written just once). This is the set of session public key of your validator.
+
+After generating the set of keys, you have to register them in the blockchain, so that they are available to all the nodes in network. You can achieve this submitting a specific extrinsic through PolkadotJS. Navigate to section `Developer` then subsection `Extrinsics` and select `session`, `setKeys` in the two dropdown panels; remember to select your validator account as `using the selected account`, then fill in the textboxes `keys: NhRuntimeSessionKeys` and `proof: Bytes` respectively with the set of session public keys you just prepared and with empty value `0x`, finally click on `Submit Transaction` button:
+
+![alt_text](./img/polkadotjs_setkeys.png)
+
+insert your account password and confirm by clicking on button `Sign and Submit`:
+
+![alt_text](./img/polkadotjs_setkeys_s1.png)
+
+in few seconds you should receive a feedback by a popup message on the top-right corner confirming you the extrinsic has been succesfully submitted.
+
+Now that the blockchain knows those public session keys are associated to your validator account, you can proceed staking some of the tokens you own in order to have a chance to be elected as a validator for the next sessions. To achieve this you have to submit another type of extrinsic.
+
+But before doing this, you'd want to know what is the current stake of other validators, in order to stake sufficient tokens to become an active validator; current **zkVerify** implementation requires to be at least in the **top 10 stakers** in order to be included in the active validators set. Navigate to section `Developer` then subsection `Chain state` and select `staking`, `erasStakersOverview` in the two dropdown panels, finally click on `+` button:
```

Comment:
- instructions could be more concise:
  - select `staking` in the `select stake query` drop-down
  - enter and select `erasStakersOverview` in the field right of the drop-down
  - "include option" has to be deselected for "+" to be clickable

Reply:
You are right, I didn't notice `include option` checkbox has to be deselected for button `+` to be clickable.
I have updated instructions accordingly.